jtuples/src/test/java/org/jtuples/TuplesTest.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    @Test
+    public void testAssociateFirst() {
+        Pair<String, Pair<String, String>> pair = new Pair<>(
+                "1",
+                new Pair<>("2", "3")
+        );
+        Tuple expected = new Pair<>(
+                new Pair<>("1", "2"),
+                "3"
+        );
+        assertEquals(expected, Tuples.associateFirst(pair));
+    }
+
+    @Test
+    public void testAssociateSecond() {
+        Pair<Pair<String, String>, String> pair = new Pair<>(
+                new Pair<>("1", "2"),
+                "3"
+        );
+        Tuple expected = new Pair<>(
+                "1",
+                new Pair<>("2", "3")
+        );
+        assertEquals(expected, Tuples.associateSecond(pair));
+    }
+
+    @Test
+    public void testMap_Pair() {
+        Pair<String, String> pair = new Pair<>("1", "2");
+        Tuple expected = new Pair<>("11", "22");
+        assertEquals(expected, Tuples.map(pair, s -> s + s));
+    }
+
+    @Test
+    public void testMap_Triple() {
+        Triple<String, String, String> triple = new Triple<>("1", "2", "3");
+        Tuple expected = new Triple<>("11", "22", "33");
+        assertEquals(expected, Tuples.map(triple, s -> s + s));
+    }
+
+    @Test
+    public void testMap_Quadruple() {
+        Quadruple<String, String, String, String> quadruple =
+                new Quadruple<>("1", "2", "3", "4");
+        Tuple expected = new Quadruple<>("11", "22", "33", "44");
+        assertEquals(expected, Tuples.map(quadruple, s -> s + s));
+    }
+
+    @Test
+    public void testMap_Quintuple() {
+        Quintuple<String, String, String, String, String> quintuple =
+                new Quintuple<>("1", "2", "3", "4", "5");
+        Tuple expected = new Quintuple<>("11", "22", "33", "44", "55");
+        assertEquals(expected, Tuples.map(quintuple, s -> s + s));
+    }
+
+    @Test
+    public void testZip_Iterable() {
+        List<String> firsts = Arrays.asList("1", "2", "3", "4");
+        List<Integer> seconds = Arrays.asList(1, 2, 3, 4);
+        List<Pair<String, Integer>> expected = Arrays.asList(
+                new Pair<>("1", 1),
+                new Pair<>("2", 2),
+                new Pair<>("3", 3),
+                new Pair<>("4", 4)
+        );
+        assertArrayEquals(expected.toArray(),
+                Tuples.zip(firsts, seconds).toArray());
+    }
+
+    @Test
+    public void testZip_Iterator() {
+        List<String> firsts = Arrays.asList("1", "2", "3", "4");
+        List<Integer> seconds = Arrays.asList(1, 2, 3, 4);
+        List<Pair<String, Integer>> expected = Arrays.asList(
+                new Pair<>("1", 1),
+                new Pair<>("2", 2),
+                new Pair<>("3", 3),
+                new Pair<>("4", 4)
+        );
+        
+        Iterator<Pair<String, Integer>> iterator =
+                Tuples.zip(firsts.iterator(), seconds.iterator());
+        
+        expected.forEach(pair -> assertEquals(pair, iterator.next()));
+    }
+    
```

Comment:
_It is obviously not the case, but anyway, just for theoretical correctness..._
What if `iterator` has more elements than `expected`? The assertions do not cover that case.